src/device.rs

```diff
@@ -81,76 +172,101 @@ impl Device {
         libevdev_set_uniq
     );
 
-    /// Returns the file associated with the device
-    ///
-    /// if the `set_fd` hasn't been called yet then it return `None`
-    pub fn fd(&self) -> Option<File> {
-        let result = unsafe { raw::libevdev_get_fd(self.raw) };
+    product_getter!(
+        product_id,
+        libevdev_get_id_product,
+        vendor_id,
+        libevdev_get_id_vendor,
+        bustype,
+        libevdev_get_id_bustype,
+        version,
+        libevdev_get_id_version
+    );
```

Comment:
Any reason for not including these functions as part of the wrapper?

Reply:
I didn't include them in `LibevdevWrapper` `trait` because UInputDevice uses these macros too. For them to work with UInputDevice, they need to use self.raw and for them to work with generic LibevdevWrapper, they need to use self.raw(), the method.

Reply:
I think it is better to avoid this duplication of code. And to use `self.raw()` at all places including the methods of `Device` and `UninitDevice`. But it can always come as a separate change. This pull request has become pretty big already so I will push it for now and we can deal with cleanup later.